Octokit.Reactive/Helpers/ConnectionExtensions.cs

```diff
@@ -28,6 +28,15 @@ public static IObservable<T> GetAndFlattenAllPages<T>(this IConnection connectio
             return GetPages(url, parameters, (pageUrl, pageParams) => connection.Get<List<T>>(pageUrl, pageParams, null).ToObservable());
         }
 
+        public static IObservable<T> GetAndFlattenAllPages<T>(this IConnection connection, Uri url, IDictionary<string, string> parameters, ApiOptions options)
```

Comment:
> I wonder if there is a cleaner way to rationalise all the overloads here to just call through to one implementation method

I think this is achievable if we make this new `GetPagesWithOptions<T>` method the endpoint. I'll call out in other places how we can achieve this.